pymess/models/common.py

```diff
@@ -29,6 +34,48 @@ def create_from_related_objects(self, *related_objects):
             for related_object in related_objects
         ]
 
+    def _get_related_object_pks(self, *related_objects):
+        return [related_object.pk for related_object in related_objects]
+
+    def _get_related_objects_qs_kwargs(self, *related_objects):
+        return reduce(OR, (
+            Q(**{
+                'object_id': obj.pk,
+                'content_type': ContentType.objects.get_for_model(obj)
+            }) for obj in related_objects
+        ))
+
+    def update_related_objects(self, *related_objects, limit_to_model=None):
+        delete_qs = self.all()
+        if limit_to_model is not None:
+            delete_qs = delete_qs.filter(content_type=ContentType.objects.get_for_model(limit_to_model))
+        if related_objects:
+            delete_qs = delete_qs.exclude(self._get_related_objects_qs_kwargs(*related_objects))
+        delete_qs.delete()
+        return self.get_or_create_from_related_objects(*related_objects)
+
+    def get_or_create_from_related_object(self, related_object):
+        return self.get_or_create(
+            object_id_int=related_object.pk if has_int_pk(related_object) else None,
+            object_id=related_object.pk,
+            content_type=ContentType.objects.get_for_model(related_object)
+        )
+
+    def get_or_create_from_related_objects(self, *related_objects):
+        return [
+            self.get_or_create_from_related_object(related_object)
+            for related_object in related_objects
+        ]
+
+    def get_related_objects_by_model(self, model):
+        pks = self.filter(content_type=ContentType.objects.get_for_model(model)).values_list('object_id_int', flat=True)
+        return model.objects.filter(pk__in=pks)
+
+    def filter_from_related_objects(self, *related_objects):
+        return self.model.objects.filter(
+            Q(template=self.instance) | Q(template__isnull=True)
```

Comment:
self.instance is set only if queryset is used via related object (template). Note for me..